src/util/async/Operation.hpp

```diff
@@ -120,6 +120,16 @@ struct BasicScheduledOperation {
     {
         state_->get().requestStop();
     }
+
+    void
+    abort() noexcept
+    {
+        if constexpr (SomeCancellableOperation<OpType>)
+            timer_.cancel();
+
+        if constexpr (SomeCancellableOperation<OpType>)
```

Comment:
Shouldn't be here a different condition?
If yes, why tests didn't catch it?

Reply:
Good catch, thanks. 
The test is checking similar logic in AnyOperation, not this one. I'll see if a test can be added 👍